### PR TITLE
Allow repocop and obligatron to read all tables

### PR DESCRIPTION
**Pull request by NovemberTang**
## What does this change?

Rather than creating a new migration every time obligatron or repocop needs access to a table, let's grant read only access to all tables in the DB

## Why?

Creating a migration is a manual, infrequent task, which means it takes a while, and is prone to mistakes. Allowing repocop and obligatron to read all tables in the DB up front eliminates this task, in exchange for minimal additional risk. No additional write permissions were granted

## How has it been verified?

Verified locally by creating the DB, running the migration, and running the SQL "unit tests" in `sql/ci.sql`. CI should also verify this.

Once approved, I will manually deploy to CODE to apply the migration, then merge.

### CODE REVIEW

File: packages/common/prisma/migrations/20240703133028_obligatron_full_read/migration.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- Allow repocop and obligatron read access to all current and future table.
+ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT ON TABLES TO repocop;
+
+GRANT SELECT ON ALL TABLES IN SCHEMA public TO obligatron;
```
Comment (akash1810) [inline packages/common/prisma/migrations/20240703133028_obligatron_full_read/migration.sql:4]: Why is this `GRANT SELECT` needed for the `obligatron` user, but not the `repocop` user?

**Comment by akash1810** [comment]
> Once approved, I will manually deploy to CODE to apply the migration, then merge.

Why wouldn't we use the automation here?

**Comment by NovemberTang** [comment]
> > Once approved, I will manually deploy to CODE to apply the migration, then merge.
> 
> Why wouldn't we use the automation here?

Sorry I meant manually deploy, as we don't have CD on CODE!

File: packages/common/prisma/migrations/20240703133028_obligatron_full_read/migration.sql (hunk 1 above)
Comment (NovemberTang) [reply packages/common/prisma/migrations/20240703133028_obligatron_full_read/migration.sql:4]: Repocop already had GRANT SELECT applied during the `repocop_user` migration. And the obligatron change wouldn't work without it, so I guess it's necessary!

File: packages/common/prisma/migrations/20240703133028_obligatron_full_read/migration.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-- Allow repocop and obligatron read access to all current and future table.
```
Comment (tjsilver) [inline packages/common/prisma/migrations/20240703133028_obligatron_full_read/migration.sql:1]: nit: "current and future tables."

**Review by akash1810 — APPROVED** [review]

**Review by tjsilver — APPROVED** [review]
Tested locally too and behaved as described.

**Pull request merged**